src/core/qgslayerdefinition.h

```diff
@@ -139,9 +140,17 @@ class CORE_EXPORT QgsLayerDefinition
         //! Whether some dependency is missing
         bool hasMissingDependency() const { return mHasMissingDependency; }
 
+        /**
+         * Returns whether the layer associated with the DOM node \a layerNode is dependent from another layer
+         *
+         * \since QGIS 3.32
+         */
+        bool isDependent( const QDomNode &layerNode ) const;
```

Comment:
I would suggest:
- rename to `isLayerDependent()`
- change the API so that layer ID is passed here, rather than DOM node (and it would be caller's responsibility in QgsProject to extract layer ID)